src/sympc/module/nn/functional.py

```diff
@@ -16,14 +16,24 @@ def relu(x: MPCTensor) -> MPCTensor:
     return res
 
 
-def mse_loss(pred: MPCTensor, target: MPCTensor) -> MPCTensor:
+def mse_loss(pred: MPCTensor, target: MPCTensor, reduction: str = "mean") -> MPCTensor:
     """Mean Squared Error loss.
 
     Args:
         pred (MPCTensor): The predictions obtained
         target (MPCTensor): The target values
+        reduction (str): the reduction method, default is `mean`
 
     Returns:
         The loss
+
+    Raises:
+        ValueError: If `reduction` not in supported methods
     """
-    return (pred - target).pow(2).sum()
+    if reduction == "mean":
+        result = (pred - target).pow(2).sum() * 1.0 / len(pred)
```

Comment:
Q: Why multiply by 1.0?
Why not use size rather than len?

Reply:
the sum maybe a int, but the mean should be flout

Reply:
in MPC it should still be an integer

Reply:
I will fix with your advice, but can you explain to me with more detail?

Reply:
Sure.

We have the `MPCTensor` class which holds tensor to `ShareTensor`s (the secret value is split across multiple parties which holds those `ShareTensors`). `ShareTensor`s are in `FixedPointPrecision` (integers).

When we use this `encoding` with `FixedPointPrecision` we transform numbers like the following:
`123.23232` ---> `12323` (this happens if we use a base of 10 and a precision of 2).

All our values are encoded and they are decoded back when we want to reconstruct the result of the computation.

Reply:
Thanks, I update the fix